Content.Server/GameTicking/Rules/ThiefRuleSystem.cs

```diff
@@ -0,0 +1,194 @@
+    [Dependency] private readonly AntagSelectionSystem _antagSelection = default!;
+    [Dependency] private readonly AudioSystem _audio = default!;
+    [Dependency] private readonly MindSystem _mindSystem = default!;
+    [Dependency] private readonly SharedRoleSystem _roleSystem = default!;
+    [Dependency] private readonly SharedJobSystem _jobs = default!;
+    [Dependency] private readonly ObjectivesSystem _objectives = default!;
+
+    const string bigObjectiveGroup = "ThiefBigObjectiveGroups";
+    const string smallObjectiveGroup = "ThiefObjectiveGroups";
+    const string escapeObjectiveGroup = "ThiefEscapeObjectiveGroups";
+
+    private const float BigObjectiveChance = 0.7f;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RulePlayerJobsAssignedEvent>(OnPlayersSpawned);
+
+        SubscribeLocalEvent<ThiefRoleComponent, GetBriefingEvent>(OnGetBriefing);
+        SubscribeLocalEvent<ThiefRuleComponent, ObjectivesTextGetInfoEvent>(OnObjectivesTextGetInfo);
+    }
+
+    private void OnPlayersSpawned(RulePlayerJobsAssignedEvent ev)
+    {
+        var query = EntityQueryEnumerator<ThiefRuleComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uid, out var thief, out var gameRule))
+        {
+            //Chance to not lauch gamerule
+            if (!_random.Prob(thief.RuleChance))
+            {
+                RemComp<ThiefRuleComponent>(uid);
+                continue;
+            }
+
+            if (!GameTicker.IsGameRuleAdded(uid, gameRule))
+                continue;
+
+            foreach (var player in ev.Players)
+            {
+                if (!ev.Profiles.ContainsKey(player.UserId))
+                    continue;
+
+                thief.StartCandidates[player] = ev.Profiles[player.UserId];
+            }
+            DoThiefStart(thief);
+        }
+    }
+
+    private void DoThiefStart(ThiefRuleComponent component)
+    {
+        if (!component.StartCandidates.Any())
+        {
+            Log.Error("There are no players who can become thieves.");
+            return;
+        }
+
+        var startThiefCount = Math.Min(component.MaxAllowThief, component.StartCandidates.Count);
+        var thiefPool = _antagSelection.FindPotentialAntags(component.StartCandidates, component.ThiefPrototypeId);
+        //TO DO: When voxes specifies are added, increase their chance of becoming a thief by 4 times >:)
+        var selectedThieves = _antagSelection.PickAntag(_random.Next(1, startThiefCount), thiefPool);
+
+        foreach(var thief in selectedThieves)
+        {
+            MakeThief(thief);
+        }
+    }
+
+    public bool MakeThief(ICommonSession thief)
+    {
+        var thiefRule = EntityQuery<ThiefRuleComponent>().FirstOrDefault();
+        if (thiefRule == null)
+        {
+            GameTicker.StartGameRule("Thief", out var ruleEntity);
+            thiefRule = Comp<ThiefRuleComponent>(ruleEntity);
+        }
+
+        //checks
```

Comment:
You should pass in ThiefRule to this method instead of assigning it to a random gamerule.

Reply:
I don't understand what's wrong here. There's no reference to random gamerules, there's only thiefrule system